### PR TITLE
Avoid using X509Certificate2's IntPtr ctor in SecureChannel

### PR DESCRIPTION
**Pull request by stephentoub**
Fixes https://github.com/dotnet/corefx/issues/37516

cc: @bartonjs

### CODE REVIEW

File: src/System.Net.Security/tests/FunctionalTests/SslStreamStreamToStreamTest.cs [hunk 1]
```diff
@@ -21,27 +21,65 @@ public abstract class SslStreamStreamToStreamTest
...
         protected abstract Task<int> ReadAsync(Stream stream, byte[] buffer, int offset, int count, CancellationToken cancellationToken = default);
 
         protected abstract Task WriteAsync(Stream stream, byte[] buffer, int offset, int count, CancellationToken cancellationToken = default);
 
-        [Fact]
-        public async Task SslStream_StreamToStream_Authentication_Success()
+        public static IEnumerable<object[]> SslStream_StreamToStream_Authentication_Success_MemberData()
         {
-            VirtualNetwork network = new VirtualNetwork();
+            yield return new object[] { null };
```
Comment (bartonjs) [inline src/System.Net.Security/tests/FunctionalTests/SslStreamStreamToStreamTest.cs]: Might be worth a comment that this means it falls back to Configuration.Certificates.GetServerCertificate() eventually.

Though I wonder why it doesn't just `yield return cert` in the using either before or after the base class version.

File: src/System.Net.Security/tests/FunctionalTests/SslStreamStreamToStreamTest.cs [hunk 2]
```diff
@@ -21,27 +21,65 @@ public abstract class SslStreamStreamToStreamTest
...
+            using (var client = new SslStream(new VirtualNetworkStream(network, isServer: false), false, AllowAnyServerCertificate))
+            using (var server = new SslStream(new VirtualNetworkStream(network, isServer: true)))
             {
-                await DoHandshake(client, server);
+                await DoHandshake(client, server, serverCert);
                 Assert.True(client.IsAuthenticated);
                 Assert.True(server.IsAuthenticated);
             }
+
+            serverCert?.Dispose();
```
Comment (bartonjs) [inline src/System.Net.Security/tests/FunctionalTests/SslStreamStreamToStreamTest.cs]: This seems better in the MemberData provider itself... unless the MemberData drains (and would have called Dispose) before executing things.

Comment (stephentoub) [reply src/System.Net.Security/tests/FunctionalTests/SslStreamStreamToStreamTest.cs]: > unless the MemberData drains (and would have called Dispose) before executing things.

That's what happens.  All of the inputs are retrieved/computed first.

File: src/System.Net.Security/tests/FunctionalTests/SslStreamStreamToStreamTest.cs (hunk 1 above)
Comment (stephentoub) [reply src/System.Net.Security/tests/FunctionalTests/SslStreamStreamToStreamTest.cs]: Changed it to return a cert

**Comment by stephentoub** [comment]
> If there are any tests which also do client authentication, it seems like they should also do a X509Certificate/X509Certificate2 variance test.

Done.

**Pull request merged**